server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -100,20 +105,30 @@ public AllocationService(
         ClusterInfoService clusterInfoService,
         SnapshotsInfoService snapshotsInfoService
     ) {
-        this(allocationDeciders, shardsAllocator, clusterInfoService, snapshotsInfoService);
+        this(allocationDeciders, shardsAllocator, clusterInfoService, snapshotsInfoService, NoopMetricsRegistry.INSTANCE);
```

Comment:
@Gaganjuneja  do we have plans to wire MetricsRegistry in all core classes?

Reply:
It's available as an guice depndency and can be injected on need basis.